astronet/direct_tensor/predict.py

```diff
@@ -0,0 +1,382 @@
+        "n_folds": len(set(fold_num)),
+        "n_points": len(fold_num),
+    }
+    for feature, value in scalar_features.items():
+        assert not np.isnan(
+            value
+        ), f"Bad nan feature for Astro ID {tce['Astro ID']}: {feature}"
+
+    all_features.update(
+        {feature: np.array([value]) for feature, value in scalar_features.items()}
+    )
+
+    return all_features
+
+
+def prepare_input(
+    feature_cfg: dict,
+    tce: pd.Series,
+    get_lc: LCGetter,
+    mode: Literal["triage", "vetting"],
+) -> dict:
+    """Assemble input features for TCE and normalize values where necessary."""
+    time, flux = get_lc(tce["Astro ID"])
+    aperture_fluxes = {}
+    if mode == "vetting":
+        aperture_fluxes = {
+            aperture: get_lc(tce["Astro ID"], aperture) for aperture in ["s", "m", "l"]
+        }
+    tce_features = prediction_features(tce, time, flux, aperture_fluxes)
+    tce_features = {
+        name: value for name, value in tce_features.items() if name in feature_cfg
+    }
+
+    features = {}
+    assert set(tce_features.keys()) == set(feature_cfg.keys())
+    for name, value in tce_features.items():
+        cfg = feature_cfg[name]
+        if not cfg["is_time_series"]:
+            if cfg.get("scale", None) == "log":
+                value = tf.cast(value, tf.float64)
+                value = tf.clip_by_value(value, cfg["min_val"], cfg["max_val"])
+                value = value - cfg["min_val"] + 1
+                value = tf.math.log(value) / tf.math.log(
+                    tf.constant(cfg["max_val"], tf.float64)
+                )
+                value = tf.cast(value, tf.float32)
+            elif cfg.get("scale", None) == "norm":
+                value = (value - cfg["mean"]) / cfg["std"]
+        features[name.lower()] = value
+
+    return {k: [v] for k, v in features.items()}
+
+
+def build_dataset(
+    feature_cfg: dict,
+    tces: pd.DataFrame,
+    get_lc: LCGetter,
+    mode: Literal["triage", "vetting"],
+    nprocs: int = 1,
+) -> tf.data.Dataset:
+    """Create Dataset object containing input tensors for all tces."""
+    tasks = [(feature_cfg, tce.to_dict(), get_lc, mode) for _, tce in tces.iterrows()]
+    if nprocs == 1:
+        all_tces_features = starmap(prepare_input, tasks)
+    else:
+        with Pool(nprocs) as pool:
+            all_tces_features = pool.starmap(prepare_input, tasks)
+    feature_table = pd.DataFrame(all_tces_features)
+    dataset = feature_table.to_dict(orient="list")
+    return tf.data.Dataset.from_tensor_slices(dataset)
+
+
+def find_checkpoints(base_dir: Path, nruns: Optional[int] = None) -> list[Path]:
+    """Find model directories, assuming base directory structure of model training checkpoints."""
+    if nruns is None:
+        nruns = len(list(base_dir.iterdir()))
+    return [next((base_dir / str(i)).iterdir()) for i in range(1, nruns + 1)]
+
+
+def batch_predict(
+    checkpoints_dir: Path,
```

Comment:
It's probably a good idea to explicitly cast `checkpoints_dir` to `Path` at some point (here or in `find_checkpoints`), since other parts of the codebase use `os.path` so astronet users may want to use string paths rather than `Path` objects.